web-portal/components/DateTimePicker.vue

```diff
@@ -263,14 +263,15 @@
         this.edit_mode = true
         this.time_segment_index = which_segment
         const time_segment = this.value[which_segment]
-        this.picker = moment(time_segment.start_time).format('YYYY-MM-DD')
-        this.start_hour = moment(time_segment.start_time).format('hh')
-        this.start_minute = moment(time_segment.start_time).format('mm')
-        this.start_ampm = moment(time_segment.start_time).format('a')
-
-        this.end_hour = moment(time_segment.end_time).format('hh')
-        this.end_minute = moment(time_segment.end_time).format('mm')
-        this.end_ampm = moment(time_segment.end_time).format('a')
+        this.picker = momenttz(time_segment.start_time).tz(time_segment.timezone).format('YYYY-MM-DD')
+        this.start_hour = momenttz(time_segment.start_time).tz(time_segment.timezone).format('hh')
+        this.start_minute = momenttz(time_segment.start_time).tz(time_segment.timezone).format('mm')
+        this.start_ampm = momenttz(time_segment.start_time).tz(time_segment.timezone).format('a')
+
+        this.end_hour = momenttz(time_segment.end_time).tz(time_segment.timezone).format('hh')
+        this.end_minute = momenttz(time_segment.end_time).tz(time_segment.timezone).format('mm')
+        this.end_ampm = momenttz(time_segment.end_time).tz(time_segment.timezone).format('a')
+        this.event_timezone = time_segment.timezone
```

Comment:
I wonder if this would be easier to read if the `momenttz(...).tz(...)` part was saved to a variable, and each subsequent assignment was just e.g. `time.format(...)`?

Might also improve performance, although we don't have any particular reason to believe it's a problem.

Reply:
great idea, will do 💡